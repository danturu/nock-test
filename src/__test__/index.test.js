const supertest = require('supertest');
const app = require('../index');

const nock = require('nock');

nock.disableNetConnect();
nock.enableNetConnect('127.0.0.1');

nock.back.fixtures = `${process.cwd()}/test/cassettes`;
nock.back.setMode('record');

const request = supertest(app.listen());

describe('index', () => {
  test('records the response', async () => {
    // Fails too: TypeError: Cannot read property 'status' of undefined
    // ...and the recorded fixture is an empty array.
    /* await new Promise((resolve, reject) => {
      nock.back('fixture.json', async (done) => {
        try {
          const { body } = await request.get('/test').set('Accept', 'application/json').set('Content-Type', 'application/json').expect(200)

          expect(body).toMatchSnapshot();

          resolve();
        } catch (err) {
          reject(err);
        }

        done();
      });
    }); */

    const { body } = await request.get('/test').set('Accept', 'application/json').set('Content-Type', 'application/json').expect(200)

    expect(body).toMatchSnapshot();
  });
});