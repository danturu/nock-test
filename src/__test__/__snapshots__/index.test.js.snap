// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index records the response 1`] = `
Object {
  "key": "value",
}
`;
